File: JavaScript/2-context.js
'use strict';

let cities = ['Athens', 'Roma', 'London', 'Beijing', 'Kiev', 'Riga'];
let f = s => s.length;

function f1() {
  let cities = ['Athens', 'Roma'];
  let f = s => s.toUpperCase();
  console.dir({ cities });
  console.dir(cities.map(f));

  {
    let f = s => s.toLowerCase();
    console.dir({ cities });
    console.dir(cities.map(f));
  }

  {
    let cities = ['London', 'Beijing', 'Kiev'];
    console.dir({ cities });
    console.dir(cities.map(f));
  }

}

f1();

console.dir({ cities });
console.dir(cities.map(f));

File: Python/2-context.py
"""Context examples"""

cities = ['Athens', 'Roma', 'London', 'Bejing', 'Kiev', 'Riga']
f = lambda s: len(s)

def f1():
    cities = ['Athens', 'Roma']
    f = lambda s: [x.upper() for x in s]
    print("in f1:")
    print(cities)
    print(f(cities))

    def inner():
        f = lambda s: [x.lower() for x in s]
        print("inner in f1:")
        print(cities)
        print(f(cities))
    inner()

f1()

print("outer:")
print(cities)
print(f(cities))
